zen/view.c

```diff
@@ -67,28 +83,39 @@ zn_view_handle_commit(struct wl_listener *listener, void *data)
 
   // FIXME: add damages of synced subsurfaces
 
-  if (!self->resize_status.resizing) {
-    return;
-  }
+  const uint32_t last_serial = self->impl->get_current_configure_serial(self);
 
-  if (!(self->resize_status.edges & (WLR_EDGE_LEFT | WLR_EDGE_TOP))) {
-    return;
+  if (self->maximize_status.changed_serial == last_serial) {
+    self->maximize_status.changed_serial = 0;
+    if (self->maximize_status.maximized) {
+      zn_view_move(self, self->board, 0, 0);
+    } else {
+      zn_view_move(self, self->board, self->maximize_status.reset_box.x,
+          self->maximize_status.reset_box.y);
+    }
   }
 
-  struct wlr_surface *surface = data;
-  int dx = 0, dy = 0;
-  if (self->resize_status.edges & WLR_EDGE_LEFT) {
-    dx = surface->previous.width - surface->current.width;
-  }
-  if (self->resize_status.edges & WLR_EDGE_TOP) {
-    dy = surface->previous.height - surface->current.height;
-  }
-  zn_view_move(self, self->board, self->x + dx, self->y + dy);
+  if (self->resize_status.resizing &&
+      self->resize_status.edges & (WLR_EDGE_LEFT | WLR_EDGE_TOP)) {
+    struct wlr_surface *surface = data;
+    int dx = 0, dy = 0;
+    if (self->resize_status.edges & WLR_EDGE_LEFT) {
+      dx = surface->previous.width - surface->current.width;
+    }
+    if (self->resize_status.edges & WLR_EDGE_TOP) {
+      dy = surface->previous.height - surface->current.height;
+    }
+    zn_view_move(self, self->board, self->x + dx, self->y + dy);
 
-  if (self->impl->get_current_configure_serial(self) ==
-      self->resize_status.last_serial) {
-    self->resize_status.resizing = false;
+    if (self->resize_status.last_serial == last_serial) {
+      self->resize_status.resizing = false;
+    }
   }
+
+  self->prev_surface_fbox.x = window_geom.x;
+  self->prev_surface_fbox.y = window_geom.y;
```

Comment:
In view-local coordinates, I think `prev_surface_fbox` should be

```suggestion
  self->prev_surface_fbox.x = -window_geom.x;
  self->prev_surface_fbox.y = -window_geom.y;
```